Osiris/Hacks/EnginePrediction.cpp

```diff
@@ -0,0 +1,44 @@
+#include "../SDK/GlobalVars.h"
+#include "../SDK/UserCmd.h"
+#include "EnginePrediction.h"
+#include "../SDK/Prediction.h"
+#include "../Interfaces.h"
+#include "../SDK/PseudoMd5.h"
+
+static float m_flOldCurtime;
+static float m_flOldFrametime;
+MoveData m_MoveData;
+
+void PredictionSys::RunEnginePred(UserCmd* cmd) noexcept
```

Comment:
In my opinion, you should let the user enable or disable this feature.

``if (!config.misc.enginePrediction) return;``

Same for [ **void PredictionSys::EndEnginePred() noexcept** ].

You will need to edit **Config.h**, **Config.cpp** and **GUI.cpp** files.

Reply:
why??? we want engine prediction on all the time is a plus for the cheat

Reply:
Maybe some people don't want it all the time. I just gave an idea.

Reply:
after reading mdlai explaination i think is best to have it on all the time

Reply:
Actually there is no reason to disable prediction and you will not find any cheats which allow you to disable prediction. It's just useless.